python_modules/dagster/dagster/_utils/schedules.py

```diff
@@ -54,20 +47,7 @@ def _exact_match(
     if cron_expression == "0 * * * *":
         return dt.minute == 0 and dt.second == 0 and dt.microsecond == 0
 
-    return (
-        dt.timestamp()
-        == _find_schedule_time(
-            cron_expression,
-            minutes,
-            hour,
-            day,
-            day_of_week,
-            schedule_type,
-            dt.add(seconds=1),
-            ascending=False,
-            already_on_boundary=False,
-        ).timestamp()
-    )
```

Comment:
what's your confidence level that this wasn't doing an important optimization for perf? given that we call _find_schedule_time below and no longer go through croniter most of the time I think it's pretty likely that it is not important. We have an AMP test suite in internal that we should be sure to run this through (it will automatically if you rebase i think)

Reply:
Yeah basically it called `_find_schedule_time` twice if it wasn't one of the simple cases. Once here to do the comparison (where the right hand side is the next boundary) and once again in the calling function to the next boundary. There's the small chance that it gets called once when the equality succeeds but it's like the opposite of an optimization in that case.